src/celeritas/global/alongstep/AlongStepRZMapFieldMscAction.cc

```diff
@@ -92,27 +92,28 @@ void AlongStepRZMapFieldMscAction::execute(CoreParams const& params,
                 std::forward<decltype(execute_track)>(execute_track)));
     };
 
-    if (this->has_msc())
-    {
-        launch_impl(
-            MscStepLimitApplier{UrbanMsc{msc_->ref<MemSpace::native>()}});
-    }
-    launch_impl(PropagationApplier{
-        RZMapFieldPropagatorFactory{field_->ref<MemSpace::native>()}});
-    if (this->has_msc())
-    {
-        launch_impl(MscApplier{UrbanMsc{msc_->ref<MemSpace::native>()}});
-    }
-    launch_impl(detail::TimeUpdater{});
-    if (this->has_fluct())
-    {
-        launch_impl(ElossApplier{FluctELoss{fluct_->ref<MemSpace::native>()}});
-    }
-    else
-    {
-        launch_impl(ElossApplier{MeanELoss{}});
-    }
-    launch_impl(TrackUpdater{});
+    launch_impl([&](CoreTrackView const& track) {
+        if (this->has_msc())
+        {
+            MscStepLimitApplier{UrbanMsc{msc_->ref<MemSpace::native>()}}(track);
+        }
+        PropagationApplier{RZMapFieldPropagatorFactory{
+            field_->ref<MemSpace::native>()}}(track);
```

Comment:
There should be zero overhead for constructing these helper classes, but it would improve readability a bit to construct the "appliers" outside of the launch kernel; then this could  just be `apply_propagation(track)`.

Reply:
I can construct the propagation applier and time/track updater outside, but I'm not sure how much it will help readability if I can't pull out the msc/eloss appliers as well.

Reply:
True. Thanks!